pagopa/io-app
# chore: [IOBP-438] Add abort dialog for payment flow

## Short description
This PR adds an abort dialog to exit the payment flow.
The dialog is displayed only if the user has already created a transaction (and locked the debt position) and, if the user confirms he wants to exit the payment flow, the transaction is deleted and the debt position unlocked.

## List of changes proposed in this pull request
- Added required locale keys
- Added `useWalletPaymentGoBackHandler` hook, which handles the dialog logic
- Added reducer and saga to handle transaction cancellation
- Fixed an issue with the header in `WalletPaymentDetailScreen`

## How to test
Using the io-dev-api-server, navigate to the Wallet Payment playground and start a new payment flow. Check that the dialog is displayed only before a transaction is created (3rd step).

## Preview

https://github.com/pagopa/io-app/assets/6160324/7f6f66e8-92f0-4581-a38a-97f10634a3ba
